### ISSUE TITLE
Fonts are 404 once consumed by other sites

### ISSUE THREAD

**Issue by alexgibson** [mozilla/protocol#49]
The Protocol `@font-face` declarations reference fonts located in `/static/`, which is a folder not copied across to NPM as it contains assets for the static docs site. This means when we pull Protocol into bedrock, the fonts are currently 404 as they reference a folder which doesn't exist. 

https://github.com/mozilla/protocol/blob/master/src/assets/sass/protocol/includes/fonts/_zilla-slab.scss#L10-L11

There are a couple of ways we could fix this:

- Copy the fonts over in our build script and switch to using relative paths. This is tidy, but may have the following issues:
  - It makes the NPM package very large as we need to bundle the fonts.
  - Sites may want to host fonts separately to their CSS & JS (e.g. a CDN, or Google Fonts).
  - Unless we're careful, we could end up loading fonts twice (e.g. global nav & footer still use Pebbles).
- The other option I can think of is not to include the `@font-face` declarations directly inside Protocol. Instead, we could have a separate file that sites could edit individually, pointing to where they host their fonts. This is more messy, but possibly more flexible as sites can edit the location.

Thoughts? /cc @craigcook @michaelmccombie

**Comment by michaelmccombie** [comment]
I agree the second option sounds like its the most flexible. I'm curious to see how other companies have handled this issue. I know that typography is a pretty common design token package as is the "tokens > framework > website" pipeline.

**Comment by alexgibson** [comment]
Ok, I took a look at what a few other folks do (such as [carbon design system](https://github.com/carbon-design-system/carbon-components), as I think there's an easy enough solution. If we use a variable in sass for the root path, then other sites can easily enough override.

In protocol:

```scss
$font-path: '/static/fonts' !default;


@font-face {
    font-family: 'Zilla Slab';
    font-weight: normal;
    font-style: normal;
    src: url('#{$font-path}/ZillaSlab-Regular.woff2') format('woff2'),
         url('#{$font-path}/ZillaSlab-Regular.woff') format('woff');
}
```

In bedrock:

```scss
$font-path: '/media/fonts';

@import "../../../node_modules/@mozilla-protocol/core/protocol/css/protocol.scss";
```

This fixes the issue for sites that want to compile their own CSS, but not for sites that may want to use the pre-compiled CSS we already supply in the protocol build. There are a couple of options:

1. We bundle the fonts in our build step and point the root path to `/node_modules/...`. We'd then need to override this in the CSS for our docs site.
2. We set the default root to some freely available URL where the fonts are already hosted, such as `https://www.mozilla.org/media/fonts/`.

Option 1. is a little messy, and I do wonder how many sites will realistically host their fonts from the `node_modules` folder. Option 2 is quite straight forward, but I do question if we should encourage people to link directly to the fonts we already host ourselves. Maybe we're OK with it, or are there reasons we should avoid it?
